Assets/HoloToolkit/Input/Scripts/Interactable.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Base class Interactable GameObjects can inherit from.
+    /// </summary>
+    public class Interactable : MonoBehaviour
+    {
+        protected bool TriggerKeywordOnGazeOnly;
+
+        protected bool IsFocused;
+
+        protected virtual void OnEnable()
+        {
+            GestureManager.Instance.OnTap += OnTap;
+            GazeManager.Instance.OnGazeExit += OnGazeExit;
+            GazeManager.Instance.OnGazeEnter += OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized += KeywordRecognized;
+        }
+
+        protected virtual void OnDisable()
+        {
+            GestureManager.Instance.OnTap -= OnTap;
+            GazeManager.Instance.OnGazeExit -= OnGazeExit;
+            GazeManager.Instance.OnGazeEnter -= OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized -= KeywordRecognized;
+        }
+
+        /// <summary>
+        /// Called when a user has tapped any gameObject.
+        /// </summary>
+        /// <param name="tappedGameObject">GameObject user has tapped.</param>
+        protected virtual void OnTap(GameObject tappedGameObject)
+        {
+            if (tappedGameObject == gameObject)
```

Comment:
I really need to make sure that we are comparing the right things.  I'm not looking at **UnityEvents** at all.   :)  That test method you are using doesn't apply.  

Here's what I'm doing:

```
using UnityEngine;
using System.Collections.Generic;
using HoloToolkit.Unity;
using System;

public class SendEventTest : Singleton<SendEventTest> {

    public delegate void OnGazeEnterEvent(GameObject focusedObject);
    public event OnGazeEnterEvent OnGazeEnter;

    List<GameObject> spawnedObjects = new List<GameObject>();
    public GameObject listenerPrefab;
    public int numObjects = 100;

    private double totalms = 0;
    // Use this for initialization
    void Start ()
    {
        numObjects = 100;
	    for(int index=0;index<numObjects;index++)
        {
            spawnedObjects.Add(Instantiate(listenerPrefab));
        }
	}
	
	// Update is called once per frame
	void Update () {
        // send the event to a random object
        DateTime start = DateTime.Now;
        GameObject rto = spawnedObjects[(int)(spawnedObjects.Count * UnityEngine.Random.value)];
        if (OnGazeEnter != null)
        {
            OnGazeEnter(rto);
        }

        totalms += (DateTime.Now - start).TotalMilliseconds;

        if (Time.frameCount % 120 == 0)
        {
            Debug.LogFormat("Frame {0}: SendEvent current total {1} ms ",Time.frameCount, totalms);
        }
    }
}


```

and

`using UnityEngine;
using System.Collections.Generic;
using System;

public class SendMessageTest : MonoBehaviour {

    List<GameObject> spawnedObjects = new List<GameObject>();
    public GameObject listenerPrefab;
    public int numObjects = 100;

    private double totalms = 0;
    // Use this for initialization
    void Start ()
    {
        numObjects = 100;
        for (int index = 0; index < numObjects; index++)
        {
            spawnedObjects.Add(Instantiate(listenerPrefab));
        }
    }
	
	// Update is called once per frame
	void Update ()
    {
        DateTime start = DateTime.Now;
        GameObject rto = spawnedObjects[(int)(spawnedObjects.Count * UnityEngine.Random.value)];
        rto.SendMessage("OnSelect", SendMessageOptions.DontRequireReceiver);
        totalms += (DateTime.Now - start).TotalMilliseconds;

        if (Time.frameCount % 120 == 0)
        {
            Debug.LogFormat("Frame {0}: SendMessage current total {1} ms ", Time.frameCount, totalms);
        }

        if (Time.frameCount % 1200 == 0)
        {
            totalms = 0;
        }
    }
}`

the prefabs referenced in the scripts have either a script that has an OnSelect() function (I tried it without too) or a script that registers for an event handler.  I measuring how much time is being taken during the call.